Improve FoldersTree background color

barijaona opened the pull request:
As recommended by AppKit documentation on controlBackgroundColor:
adds a subtle transparency effect.

Align indentation with neighboring lines

Eitot commented:
I think we should ignore any files that are automatically generated.

josh64x2 commented:
I just tested in 10.8 and the help files load 😄 (well, the 🇬🇧 ones do anyway).

josh64x2 commented:
OK it should be good to go if you want to give it a quick test with a lang other than French or English. (Might be a good idea later to add the languages that haven't been translated e.g. japanese, korean, to a "translation help wanted" post on the website).

Eitot commented:
I think I found the problem. `CFBundleHelpBookName` is supposed to be a bundle identifier, not a name. 😑 That should fix it.

I will rename Vienna.help into Vienna Help and add it as a separate Xcode bundle target instead. Then we can let Xcode generate the help bundle automatically, while using its localisation features.

josh64x2 commented:
Oh I saw that note about `CFBundleHelpBookName` as well, except it doesn't work - Apple's instructions are wrong (as mentioned here https://alastairs-place.net/blog/2015/01/14/apple-help-in-2015/) and I couldn't get my`Vienna.help` to work until I just used a name and not a bundle identifier.

josh64x2 commented:
relevant quote:
>The second thing to note is that in spite of the documentation implying that you can use your help bundle identifier to refer to your help bundle (which would, admittedly, make sense), you can’t. You need to use the HPDBookTitle value.

Eitot commented:
For me it is the opposite, now the link in preferences doesn’t work anymore. 😂

Eitot commented:
This is so bizarre. When I build your PR, it doesn’t work. When I change `CFBundleHelpBookName` to the identifier, then it does. I wonder if there is some caching involved here that causes this.

josh64x2 commented:
Haha yeah maybe delete DerivedData and try?

Eitot commented:
Nope. ☹️

josh64x2 commented:
It must be a build issue because when I copied the .app bundle to my mountain lion VM it worked...

Have you tried a clean git checkout into a different directory?

Eitot commented:
Yep, no difference. I’ll try it in a VM as well.

Eitot commented:
I have tested two deployment versions on ML and Mavericks and they both work. However, I also noticed that there is caching. I changed the font in the style sheet of one of them and the help viewer does not show it, it will keep the first style sheet instead. After a while, it is refreshed.

I think the problem may be located there. My theory is that the system indexes the help book separately and does not do this properly when there are multiple versions of the same program installed. The good news is that the bundle name now also works on Sierra.

Eitot committed:
Address several deprecation warnings

Eitot committed:
Remove unused properties

Eitot committed:
Change color of unread count

Eitot committed:
Fix toolbar item validation

`-validateToolbarItem:` is no longer implemented by NSObject by default. The previous implementation could result in an exception on at least macOS 11 Big Sur if the responder did not implement this method. The explicit protocol conformance and validation prevents this from happening.

barijaona committed:
Improve FoldersTree background color
